patches/tModLoader/Terraria/ModLoader/IO/PlayerIO.cs

```diff
@@ -347,6 +349,31 @@ internal static string SaveUsedModPack(Player player)
 		return Path.GetFileNameWithoutExtension(Core.ModOrganizer.ModPackActive);
 	}
 
+	internal static TagCompound SaveLoadouts(EquipmentLoadout[] equipLoadouts)
+	{
+		TagCompound loadouts = new();
+
+		for (int i = 0; i < equipLoadouts.Length; i++) {
+			loadouts[$"loadout{i}Armor"] = SaveInventory(equipLoadouts[i].Armor);
+			loadouts[$"loadout{i}Dye"] = SaveInventory(equipLoadouts[i].Dye);
+		}
+
+		return loadouts;
+	}
+
+	internal static void LoadLoadouts(EquipmentLoadout[] loadouts, TagCompound fullSavedCompound)
+	{
+		if (!fullSavedCompound.TryGet("loadouts", out TagCompound loadoutTag)) {
+			// This indicates that this player was created before 1.4.4, and thus has no loadouts; so nothing to load here.
+			return;
+		}
```

Comment:
This actually isn't necessary, you can just use `tag.GetCompound("loadouts")` in the caller